week2/internal/controller/http/v1/actor_handler.go

```diff
@@ -79,3 +79,65 @@ func (handler *ActorHandler) Create(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, model.HttpResponse[entity.Actor]{Message: "Success", Data: actor})
 }
+
+// @Summary Update an actor
+// @Description Update an actor
+// @Tags Actor
+// @Accept json
+// @Param id path int true "actorId"
+// @RequestBody model.ActorRequest true "Actor payload"
+// @Produce  json
+// @Router /actors/{id} [put]
+// @Success 200 {object} entity.Actor
+func (handler *ActorHandler) Update(c *gin.Context) {
+	//check param id
+	id := c.Param("id")
+	parsedId, err := strconv.ParseInt(id, 10, 64)
+	if err != nil {
+
+		c.JSON(http.StatusBadRequest, model.HttpResponse[any]{Message: err.Error(), Data: nil})
+	}
+
+	var actorRequest model.ActorRequest
+	//binding request
+	if err1 := c.ShouldBindJSON(&actorRequest); err1 != nil {
```

Comment:
as a convention, please use only 1 declaration of err through out the function. Please change all ``` err1, err2, ...  ```  to  ``` err ```